src/main/java/calculator/Numbers.java

```diff
@@ -0,0 +1,51 @@
+package calculator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+public class Numbers {
+
+    private final List<Number> numbers;
+
+    private Numbers(final List<Number> numbers) {
+        this.numbers = new ArrayList<>(Collections.unmodifiableList(numbers));
```

Comment:
List<Number>가 불변하게 유지되길 원하시고 add 등의 메소드 호출 시 `UnsupportedOperationException`이 발생하길 기대하고 위와 같은 로직을 작성해주신걸로 이해했어요.

이 부분은 확인이 필요하겠지만 `new ArrayList<>()`로 감싸는 과정에서 의도한 대로 동작하지 않을 것 같습니다.
아래와 같이 정의해야 할 것 같아요.

```suggestion
        this.numbers = Collections.unmodifiableList(numbers);
```

Reply:
피드백 감사합니다 :)
완전히 놓치고 있었던 부분 이네요 😢 
제가 의도한 코드는 List의 불변이 유지되길 원하기도 하면서 외부 참조에 의한 리스트의 변경이 이루어지지 않길 원했습니다.
아래의 테스트 코드 처럼요 ! 
<img width="435" alt="스크린샷 2020-12-08 오후 10 11 51" src="https://user-images.githubusercontent.com/44187892/101488091-7aa43880-39a2-11eb-816f-7b4fc9c6af0e.png">

하지만 제가 작성한 코드에서는 **외부 참조에 의한 리스트 변경은 막았지만**
`UnsupportedOperationException `은 의도한대로 발생되지는 않는 걸 확인했습니다 !
두 가지 부분을 동시에 만족하기는 힘든 걸까요.. 🤔

Reply:
피드백 감사합니다 :)
완전히 놓치고 있었던 부분 이네요 👍